vm/systemSmartContracts/delegation.go

```diff
@@ -1073,12 +1134,14 @@ func (d *delegation) unDelegate(args *vmcommon.ContractCallInput) vmcommon.Retur
 
 	globalFund.TotalUnStakedFromNodes.Add(globalFund.TotalUnStakedFromNodes, unStakeFromNodes)
 	activeFund.Value.Sub(activeFund.Value, actualUserUnStake)
+	//TODO: If saveFund will fail, then the save done in method resolveUnStakedUnBondResponse -> saveDelegationStatus will remain saved. Is this ok?
```

Comment:
nothing will remain. Any error invalidates all previous changes. save is done in a cache - cache is invalidated if returnCode != vmcommon.Ok

Reply:
ok